recipes/cpr/all/conanfile.py

```diff
@@ -18,12 +18,16 @@ class CprConan(ConanFile):
         "fPIC": [True, False],
         "with_openssl": [True, False],
         "with_winssl": [True, False],
+        "with_darwinssl": [True, False],
+        "no_ssl": [True, False],
```

Comment:
I was afraid of this... 

Can we copy the curl recipe and use an `with_ssl` and provide string options? and deprecate the older two?

Reply:
You know, that makes total sense, and frankly I don't know why I didn't think about it. Will do.

Reply:
Is there a proper way to deprecate options, by the way? I took a quick peek at the docs and didn't see it

Reply:
#5020 We just wrote it recently 🎉

Reply:
Oh awesome! Will flip to using that way